src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/MLDsa/MLDsaTests.cs

```diff
@@ -0,0 +1,198 @@
+
+            byte[] pubKey = pubKeyHex.HexToByteArray();
+            byte[] msg = msgHex.HexToByteArray();
+            byte[] sig = sigHex.HexToByteArray();
+            byte[] context = contextHex.HexToByteArray();
+            using MLDsa mldsa = MLDsa.ImportMLDsaPublicKey(PickAlgorithmBasedOnMLDsaPublicKeySize(pubKey.Length), pubKey);
+            Assert.True(mldsa.VerifyData(msg, sig, context));
+        }
+
+        public static IEnumerable<object[]> AllMLDsaGenerateKey()
+        {
+            yield return new object[] { (Func<MLDsa>)MLDsa.GenerateMLDsa44Key };
+            yield return new object[] { (Func<MLDsa>)MLDsa.GenerateMLDsa65Key };
+            yield return new object[] { (Func<MLDsa>)MLDsa.GenerateMLDsa87Key };
+        }
+
+        private static MLDsaAlgorithm PickAlgorithmBasedOnMLDsaSecretKeySize(int keySize)
+        {
+            return keySize switch
+            {
+                2560 => MLDsaAlgorithm.MLDsa44,
+                4032 => MLDsaAlgorithm.MLDsa65,
+                4896 => MLDsaAlgorithm.MLDsa87,
+                _ => throw new NotSupportedException($"Unsupported key size: {keySize}"),
+            };
+        }
+
+        private static MLDsaAlgorithm PickAlgorithmBasedOnMLDsaPublicKeySize(int keySize)
+        {
+            return keySize switch
+            {
+                1312 => MLDsaAlgorithm.MLDsa44,
+                1952 => MLDsaAlgorithm.MLDsa65,
+                2592 => MLDsaAlgorithm.MLDsa87,
+                _ => throw new NotSupportedException($"Unsupported key size: {keySize}"),
+            };
```

Comment:
Please don't.  This is bad precedent, and not forward-looking.

Seealso SHA-2-256 vs SHA-3-256.